test/mountain_view/component_test.rb

```diff
@@ -17,15 +17,95 @@ def test_styleguide_stubs
     component = MountainView::Component.new("header")
 
     expected_stub = {
-      id: 1,
-      title: "20 Mountains you didn't know they even existed",
-      subtitle: "Buzzfeed title"
-    }
+                      :meta=>"There is this different classes",
+                      :stubs=>
+                        [
+                          {:id=>1, :title=>"20 Mountains you didn't know they even existed",
+                           :subtitle=>"Buzzfeed title"},
+                          {:id=>2, :title=>"You won't believe what happened to this man at Aspen"}
+                        ]
+                     }
 
     assert_instance_of Array, component.styleguide_stubs
     assert_equal expected_stub, component.styleguide_stubs.first
   end
 
+  def test_component_stubs
+    component = MountainView::Component.new('header')
+    expected_stub = [
+                      {:id=>1, :title=>"20 Mountains you didn't know they even existed",
+                       :subtitle=>"Buzzfeed title"},
+                      {:id=>2, :title=>"You won't believe what happened to this man at Aspen"}
+                    ]
+
+    assert_instance_of Array, component.component_stubs
+    assert_equal expected_stub, component.component_stubs
+  end
+
+  def test_component_stubs?
+    component_with_stubs = MountainView::Component.new("header")
+    component_with_empty_stub_file = MountainView::Component.new("breadcrumbs")
+    component_without_stub_file = MountainView::Component.new("social_media_icons")
+    compoenet_with_stubs_but_incorrect_format = MountainView::Component.new('card')
+
+    assert_equal true, component_with_stubs.component_stubs?
+    assert_equal false, component_without_stub_file.component_stubs?
+    assert_equal false, component_with_empty_stub_file.component_stubs?
+    assert_equal false, compoenet_with_stubs_but_incorrect_format.component_stubs?
+  end
+
+  def test_stubs_extra_info
+    component_with_extra_info = MountainView::Component.new("header")
+    component_with_empty_stub_file = MountainView::Component.new("breadcrumbs")
+    component_without_stub_file = MountainView::Component.new("social_media_icons")
+    expected_extra_info_stub = "There is this different classes"
+
+    assert_equal expected_extra_info_stub, component_with_extra_info.stubs_extra_info
+    assert_equal nil, component_with_empty_stub_file.stubs_extra_info
+    assert_equal nil, component_without_stub_file.stubs_extra_info
+  end
+
+  def test_stubs_extra_info?
+    component_with_stubs = MountainView::Component.new("header")
+    component_with_empty_stub_file = MountainView::Component.new("breadcrumbs")
+    component_without_stub_file = MountainView::Component.new("social_media_icons")
+    compoenet_with_stubs_but_incorrect_format = MountainView::Component.new('card')
```

Comment:
Line is too long. [83/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.